### ISSUE TITLE
Listing backups and restore data from backups

### ISSUE THREAD

**Issue by yattagit** [CrunchyData/postgres-operator#1889]
Hi,
I'm using PGO 4.4.0 version and trying to backup Postgre DB cluster and restore data again.
I'm using Azure WS environment.

**1-** I'm taking backup with the command below,
`pgo backup hacluster --backup-type=pgdump --backup-opts="--dump-all"`

But when i run " pgo show backup hacluster" command, I don't see any backup list in there?

**2-** If i run the command below, command is creating "testpvc" pvc successfully.
`pgo backup hacluster --backup-type="pgdump" --pvc-name="testpvc"`

But I can't able to restore data from this PVC.
I checked whole Crunchy Data pages and tried lots of variation of commands but still same.

**Basically, which command should I use?**
(I checked the full command reference and other ref pages there is nothing about this.)

Thanks!

**Comment by jkatz** [comment]
1. [`pgo show backup hacluster --backup-type=pgdump`](https://access.crunchydata.com/documentation/postgres-operator/latest/pgo-client/reference/pgo_show_backup/)

2. What command are you using to restore? Where are you trying to restore to? Are you trying to restore data to the existing PostgreSQL cluster or are you creating a new cluster?

For your final question, I provided a detailed explanation in https://github.com/CrunchyData/postgres-operator/issues/1885#issuecomment-692827731 about the differences between the backup types and what they are typically used for. The general rule of thumb: if you're looking for something for disaster recovery purposes or want to be able to create new clusters from the backups, use physical backups. If you're looking to migrate data around or only use a subset of your data, use logical backups.

**Comment by yattagit** [comment]
Hi,
1. Thanks!

2. I'm trying "pgo restore" command to restore data from backup. I know it's destructive but still no problem for me.

Regarding to documentation, I can use "**`pgo restore mycluster  --backup-pvc="testpvc"`**" command? But it's getting nothing, data is still same.

```
PS C:\Users\user> kubectl get pvc
NAME                                   STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
backup-hacluster-pgdump-postgres-pvc   Bound    pvc-7c7183df-4666-4615-9e38-916ee4879a39   1Gi        RWO            default        34m
hacluster                              Bound    pvc-421c456b-ab35-4728-8b27-914f9a0197c7   1Gi        RWO            default        5h26m
hacluster-ihti                         Bound    pvc-2170cd94-1fad-48b5-a5ad-dc6b60c203f0   1Gi        RWO            default        5h24m
hacluster-pgbr-repo                    Bound    pvc-ff370559-c99d-4bbe-97f7-1f57ce63d67e   1Gi        RWO            default        5h26m
**testpvc**                                Bound    pvc-77d4ae58-12da-41f1-9125-6d4dabb94e3e   1Gi        RWO            default        5h17m
PS C:\Users\user>
```

**Comment by jkatz** [comment]
I still need some more guidance to make a recommendation. What are you trying to accomplish with the restore?

- Are you trying to restore your database back to a specific point-in-time?
- Are you trying to restore your data to use a particular copy of the data?
- Are you trying to create a new cluster from the backup?

**Comment by yattagit** [comment]
First one, I'm trying to restore my database back to a specific point-in-time. 
I guess there are 2 options; from a PVC or from the command's output below;

```
bash-4.2$ pgo show backup hacluster
cluster: hacluster
storage type: local
stanza: db
    status: ok
    cipher: none
    db (current)
        wal archive min/max (12-1)

        full backup: 20200915-155556F
            timestamp start/stop: 2020-09-15 15:55:56 +0000 UTC / 2020-09-15 15:56:24 +0000 UTC
            wal start/stop: 000000010000000000000002 / 000000010000000000000002
            database size: 31.1MiB, backup size: 31.1MiB
            repository size: 3.7MiB, repository backup size: 3.7MiB
            backup reference list:

        incr backup: 20200915-155556F_20200915-155957I
            timestamp start/stop: 2020-09-15 15:59:57 +0000 UTC / 2020-09-15 16:00:03 +0000 UTC
            wal start/stop: 000000010000000000000005 / 000000010000000000000005
            database size: 31.1MiB, backup size: 2.8MiB
            repository size: 3.7MiB, repository backup size: 322.2KiB
            backup reference list: 20200915-155556F

        full backup: 20200915-161628F
            timestamp start/stop: 2020-09-15 16:16:28 +0000 UTC / 2020-09-15 16:16:58 +0000 UTC
            wal start/stop: 00000001000000000000000A / 00000001000000000000000A
            database size: 31.1MiB, backup size: 31.1MiB
            repository size: 3.7MiB, repository backup size: 3.7MiB
            backup reference list:

bash-4.2$
```

**Comment by jkatz** [comment]
It sounds like you want the [restore in-place](https://access.crunchydata.com/documentation/postgres-operator/latest/architecture/disaster-recovery/#restore-in-place) method, specifically this example:

[https://access.crunchydata.com/documentation/postgres-operator/latest/pgo-client/common-tasks/#disaster-recovery-backups-restores#point-in-time-recovery-pitr-1](https://access.crunchydata.com/documentation/postgres-operator/latest/pgo-client/common-tasks/#disaster-recovery-backups-restores#point-in-time-recovery-pitr-1)

First, the "PVC" option is just for **logical** backups. For physical backups, this is managed by pgBackRest. pgBackRest does store these to their own PVC but they are interfaced differently. You do not need to reference the pgBackRest PVC.

The command you display above is exactly how you would do a PITR.

Now, if you wanted to do a point-in-time-recovery and create a *new* PostgreSQL cluster using an existing pgBackRest repository, e.g. from the pgBackRest PVC, you would reference the **name** of that PostgreSQL cluster. For instance, let's say I want to create a new PostgreSQL cluster called `hipo` that has its data at a specific point in time. Usuing your above example:

```
pgo create cluster hippo --restore-from=hacluster \
  --restore-opts="--type=time --target='2020-09-15 16:16:28 +0000'"
```

-----

Let's advance this one step further. Let's say I want to keep a "PVC" that just contains the pgBackRest repository but use it to spawn new PostgreSQL clusters. Using your example above:

```
pgo delete cluster hacluster --keep-backups
pgo create cluster hippo --restore-from=hacluster \
  --restore-opts="--type=time --target='2020-09-15 16:16:28 +0000'"
```

which achieves the net effect for "restoring from a PVC."
